impeller/entity/entity_pass.h

```diff
@@ -52,6 +52,13 @@ class EntityPass {
 
   void IterateAllEntities(const std::function<bool(Entity&)>& iterator);
 
+  /// @brief Iterates all entities on this pass, returning if there is a
+  /// subpass.
```

Comment:
Nit: "Iterate entities in this pass up until the first subpass is found. This is useful for limiting look-ahead optimizations."

Reply:
Done